utilities/xmlrpcpp/src/XmlRpcClient.cpp

```diff
@@ -431,13 +436,16 @@ XmlRpcClient::readHeader()
     return false;   // We could try to figure it out by parsing as we read, but for now...
   }
 
-  _contentLength = atoi(lp);
-  if (_contentLength <= 0) {
-    XmlRpcUtil::error("Error in XmlRpcClient::readHeader: Invalid Content-length specified (%d).", _contentLength);
+  // avoid overly large or improperly formatted content-length
+  long int clength = 0;
+  clength = strtol(lp, nullptr, 10);
```

Comment:
You should probably check errno here as well so you don't silently accept wrong things:

> On success, the function returns the converted integral number as a long int value.
If no valid conversion could be performed, a zero value is returned (0L).
If the value read is out of the range of representable values by a long int, the function returns LONG_MAX or LONG_MIN (defined in <climits>), and errno is set to ERANGE.

Reply:
Actually, it's unclear if ernno is set if no valid conversation could be performed, can you clarify?

Reply:
As I understand it errno applies only when the function returns LONG_MAX or LONG_MIN, both of which will be caught in the statement below. In any case I don't believe we can easily capture all invalid input but we can always return 0 length for invalid input. For example:

`content-length=0` -> clength=0, errno=0, valid case.
`content-length=100ABC` -> clength=100, errno=0, invalid case not captured.
`content-length=0x0D` -> clength=0, errno is unclear, invalid case not captured above but xml will not be parsed.
`content-length=aa` -> clength=0, errno may be set to EINVAL, invalid case not captured above but xml will not be parsed.

Reply:
Alright very good.